pandas/tests/indexes/interval/test_interval_new.py

```diff
@@ -127,41 +126,46 @@ def test_slice_locs_with_ints_and_floats_succeeds(self):
 
         # decreasing non-overlapping
         index = IntervalIndex.from_tuples([(3, 4), (1, 2), (0, 1)])
-        assert index.slice_locs(0, 1) == (3, 2)
-        assert index.slice_locs(0, 2) == (3, 1)
+        assert index.slice_locs(0, 1) == (3, 3)
+        assert index.slice_locs(0, 2) == (3, 2)
         assert index.slice_locs(0, 3) == (3, 1)
-        assert index.slice_locs(3, 1) == (1, 2)
-        assert index.slice_locs(3, 4) == (1, 0)
-        assert index.slice_locs(0, 4) == (3, 0)
+        assert index.slice_locs(3, 1) == (1, 3)
+        assert index.slice_locs(3, 4) == (1, 1)
+        assert index.slice_locs(0, 4) == (3, 1)
 
     @pytest.mark.parametrize("query", [
-        [0, 1], [0, 2], [0, 3], [3, 1], [3, 4], [0, 4]])
+        [0, 1], [0, 2], [0, 3], [0, 4]])
     @pytest.mark.parametrize("tuples", [
-        [(0, 2), (1, 3), (2, 4)], [(2, 4), (1, 3), (0, 2)],
-        [(0, 2), (0, 2), (2, 4)], [(0, 2), (2, 4), (0, 2)],
+        [(0, 2), (1, 3), (2, 4)],
+        [(2, 4), (1, 3), (0, 2)],
+        [(0, 2), (0, 2), (2, 4)],
+        [(0, 2), (2, 4), (0, 2)],
         [(0, 2), (0, 2), (2, 4), (1, 3)]])
     def test_slice_locs_with_ints_and_floats_errors(self, tuples, query):
+        start, stop = query
         index = IntervalIndex.from_tuples(tuples)
         with pytest.raises(KeyError):
-            index.slice_locs(query)
+            index.slice_locs(start, stop)
 
     @pytest.mark.parametrize('query, expected', [
-        ([Interval(1, 3, closed='right')], [1]),
-        ([Interval(1, 3, closed='left')], [-1]),
-        ([Interval(1, 3, closed='both')], [-1]),
-        ([Interval(1, 3, closed='neither')], [-1]),
+        ([Interval(2, 4, closed='right')], [1]),
+        ([Interval(2, 4, closed='left')], [-1]),
+        ([Interval(2, 4, closed='both')], [-1]),
+        ([Interval(2, 4, closed='neither')], [-1]),
         ([Interval(1, 4, closed='right')], [-1]),
         ([Interval(0, 4, closed='right')], [-1]),
-        ([Interval(1, 2, closed='right')], [-1]),
-        ([Interval(2, 4, closed='right'), Interval(1, 3, closed='right')],
-         [2, 1]),
-        ([Interval(1, 3, closed='right'), Interval(0, 2, closed='right')],
+        ([Interval(0.5, 1.5, closed='right')], [-1]),
+        ([Interval(2, 4, closed='right'), Interval(0, 1, closed='right')],
          [1, -1]),
-        ([Interval(1, 3, closed='right'), Interval(1, 3, closed='left')],
+        ([Interval(2, 4, closed='right'), Interval(2, 4, closed='right')],
+         [1, 1]),
+        ([Interval(5, 7, closed='right'), Interval(2, 4, closed='right')],
+         [2, 1]),
+        ([Interval(2, 4, closed='right'), Interval(2, 4, closed='left')],
          [1, -1])])
     def test_get_indexer_with_interval(self, query, expected):
 
-        tuples = [(0, 2.5), (1, 3), (2, 4)]
+        tuples = [(0, 2), (2, 4), (5, 7)]
```

Comment:
change here was to have a non-overlapping index?

Reply:
Yes, since overlapping indexes are invalid for `get_indexer` and need to use `get_indexer_non_unique` instead.  For intervals, overlapping is the analogue for uniqueness, as uniqueness isn't a strong enough condition since a query on a unique `IntervalIndex` can still have multiple values returned if they come from the overlap of 2+ intervals.